sdk/identity/Azure.Identity/src/ManagedIdentityClient.cs

```diff
@@ -27,13 +26,13 @@ public ManagedIdentityClient(CredentialPipeline pipeline, string clientId = null
         public ManagedIdentityClient(ManagedIdentityClientOptions options)
         {
             _options = options;
-            ClientId = options.ClientId;
+            ClientId = options.ClientId ?? EnvironmentVariables.GetNonEmptyStringOrNull(EnvironmentVariables.ClientId);
```

Comment:
I don't want to change this to read the environment variable. By design only the `DefaultAzureCredential` and `EnvironmentCredential` are configurable through the environment.